### ISSUE TITLE
False positive: 'Nested-interactive' rule fails on nested elements with negative tabindex

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#3231]
The ```nested-interactive``` rule fails on the following example despite the parent button working as expected with screen readers and Dragon 

    <button id="user_info_dropdown">
        <div aria-hidden="true">
            <div tabindex="-1" role="presentation">
                <span>Name</span>
            </div>
        </div>
        <span>User menu</span>
    </button>

Product: axe-core

Expectation: Not to fail ```nested-interative``` rule on this example

Motivation: I do not think there is a need for the nested elements here to be available to ATs 

<pre><code>
axe-core version:4.3.3
</pre>
